src/managed/Microsoft.DotNet.PlatformAbstractions/Native/PlatformApis.cs

```diff
@@ -68,6 +72,23 @@ private static string GetDarwinVersion()
             }
         }
 
+        private static string GetFreeBSDVersion()
+        {
+            // This is same as sysctl kern.version
+            // FreeBSD 11.0-RELEASE-p1 FreeBSD 11.0-RELEASE-p1 #0 r306420: Thu Sep 29 01:43:23 UTC 2016     root@releng2.nyi.freebsd.org:/usr/obj/usr/src/sys/GENERIC
+            // What we want is major release as minor releases should be compatible.
+            String version = RuntimeInformation.OSDescription;
+            try
+            {
+                // second token up to first dot
+                return RuntimeInformation.OSDescription.Split()[1].Split('.')[0];
```

Comment:
This looks fragile. Is the version string guaranteed to have the format you assume? It seems it would be nice to use the same method to extract the version in the C++ pal::get_current_os_rid_platform and here. That would also make the version extraction less fragile. You'd need to add a new method to NativeMethods.Unix.cs to pinvoke the `sysctlbyname` and call it from here with the "kern.osrelease"

Reply:
https://www.freebsd.org/releases/
There always should be number and dot. 
I was thinking about kern.osrelease but the OSDescription was conveniently works and also works in net45 branch (after adding proper reference)

Reply:
Ok, let's keep it this way for now.